casepropods/family_connect_subscription/plugin.py

```diff
@@ -78,6 +112,69 @@ def format_schedule(self, schedule):
             schedule['month_of_year'], schedule['day_of_week'])
         return prettify_cron(cron_schedule)
 
+    def cancel_subscriptions(self, subscription_ids):
+        params = {'active': False}
+        for subscription in subscription_ids:
+            try:
+                self.stage_based_messaging_api.update_subscription(
+                    subscription, params)
+            except HTTPServiceError:
+                return False
+        return True
+
+    def perform_action(self, type_, params):
+        if type_ == "cancel_subs":
+            subscription_ids = params.get("subscription_ids", [])
+            success = self.cancel_subscriptions(subscription_ids)
+            if not success:
+                return (False,
+                        {"message": "Failed to cancel some subscriptions"})
+            return (True, {"message": "cancelled all subscriptions"})
```

Comment:
I decided not to use one here because of how the different messages are treated on the front end.
Failure messages are put in a temporary block at the top of the page whereas success messages have the active user's name pre-pended to them and are then stored as a note on the case.